### PR TITLE
Add user_guild_settings support

### PR DESCRIPTION
**Pull request by Ratismal**
In the ready event, discord sends two properties specific to ClientUsers - UserSettings and UserGuildSettings. UserSettings has already been wrapped, so this PR aims to wrap UserGuildSettings.

This is accomplished by create two new structures: ClientUserGuildSettings and ClientUserChannelOverrides. These contain wrappings for the user guild settings, and the former contains a Collection of the latter.

Additionally, the Guild and GuildChannel classes both have new properties obtained from these wraps.

I didn't implement the actions required to modify these settings, and will look into doing that in the future (I didn't implement because I wasn't sure how to do it).

I also didn't know the best way to accomplish this, so feel free to give feedback or suggestions on reimplementation.

Thank you for your time.

### CODE REVIEW

**Comment by GusCaplan** [comment]
you could replace these waterfalls with try/catch

```js
get muted() {
  if (this.client.user.bot) return null;
  if (!this.client.user.guildSettings) return null;
  if (!this.client.user.guildSettings.get(this.guild.id)) return null;
  if (!this.client.user.guildSettings.get(this.guild.id).channelOverrides) return false;
  if (!this.client.user.guildSettings.get(this.guild.id).channelOverrides.get(this.id)) return false;
  return this.client.user.guildSettings.get(this.guild.id).channelOverrides.get(this.id).muted;
}
```
to
```js
get messageNotifications() {
  try {
    return this.client.user.guildSettings.get(this.guild.id).channelOverrides.get(this.id).muted;
  } catch (err) {
    return false;
  }
}
```

**Comment by aemino** [comment]
This needs a rebase.

**Comment by Ratismal** [comment]
As requested by @GusCaplan, switched waterfalls to try/catches

Also did some `if bot return null` checks

**Comment by GusCaplan** [comment]
@Ratismal this needs a rebase

**Comment by devsnek** [comment]
@Ratismal is this ready to merge

File: src/structures/Guild.js [hunk 1]
```diff
@@ -332,6 +332,61 @@ class Guild {
...
+  /**
+   * Whether to suppress everyone messages
+   * <warn>This is only available when using a user account.</warn>
+   * @type {boolean}
+   */
+  get suppressEveryone() {
+    if (this.client.user.bot) return null;
+    if (!this.client.user.guildSettings) return null;
+    if (!this.client.user.guildSettings.get(this.id)) return null;
+    return this.client.user.guildSettings.get(this.id).suppressEveryone;
```
Comment (devsnek) [inline src/structures/Guild.js]: try/catch

File: src/util/Constants.js [hunk 2]
```diff
@@ -230,6 +230,7 @@ exports.Events = {
   USER_UPDATE: 'userUpdate',
   USER_NOTE_UPDATE: 'userNoteUpdate',
   USER_SETTINGS_UPDATE: 'clientUserSettingsUpdate',
+  USER_GUILD_SETTINGS_UPDATE: 'clientUserGuildSettingsUpdate',
```
Comment (devsnek) [inline src/util/Constants.js]: i would change this to `guildSettingsUpdate`

**Pull request merged**